app/content/profile/index.js

```diff
@@ -125,7 +138,68 @@ function Index() {
   }
 
   const handleImageClick = () => {
-    console.log("TODO: implement and open an image upload modal!");
+    setIsImageUploadModalVisible(true);
+  };
+
+  // Function to handle closing the image upload modal
+  const hideImageUploadModal = () => {
+    setIsImageUploadModalVisible(false);
+  };
+
+  const firebaseProfileImageUpload = async (uri) => {
+    try {
+      setImageUploading(true);
+      // Fetch the image data from the URI
+      const response = await fetch(uri);
+      const blob = await response.blob();
+
+      // Get a reference to the Firebase Storage instance
+      const storage = getStorage();
+
+      // Create a reference to the storage location where the image will be stored
+      const storageRef = ref(storage, userId);
+
+      // Upload the image to Firebase Storage
+      const snapshot = await uploadBytes(storageRef, blob);
+
+      console.log("Uploaded the image blob to the Firebase storage:", snapshot);
+
+      // Get the download URL for the uploaded image
+      const downloadURL = await getDownloadURL(snapshot.ref);
+      await updateDoc(doc(db, "teams", currentTeamId, "users", userId), {
+        pfp: downloadURL,
+      });
+
+      queryClient.invalidateQueries({ queryKey: ["user", { userId }] });
+      setSnackbarMessage("Successfully uploaded the profile picture!");
+      setImageUploading(false);
+
+      return downloadURL;
+    } catch (error) {
+      console.error("Error uploading image to Firebase:", error);
+      setSnackbarMessage("Error uploading profile picture. Please try again.");
+      setImageUploading(false);
+
+      throw error; // Rethrow the error to handle it at the caller's level if needed
+    }
+  };
+
+  // Function to handle uploading image locally
+  const handleImageUpload = async () => {
+    let imageResult = await launchImageLibraryAsync({
+      mediaTypes: MediaTypeOptions.Images,
+      allowsEditing: true,
+      aspect: [4, 3],
+      quality: 1,
+    });
+
+    // console.log(imageResult);
+
+    if (!imageResult.canceled) {
```

Comment:
I did get a console warning, but it seems safe to ignore (since we are indeed using `canceled`)

![image](https://github.com/Golf-Drill-Challenge-App/Golf-App/assets/82061589/8063d43d-f5b6-4a92-911d-726ae205a5f8)

Reply:
If we did want to remove the warning, there are some tips here: https://github.com/expo/expo/issues/20977#issuecomment-1544422682